data/mods/ssb/abilities.ts

```diff
@@ -25,6 +25,32 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 			}
 		},
 	},
+	// Darth
+	guardianangel: {
+		desc: "Regenerator effect and changes type on switch in to be super effective against foe.", // long description
+		shortDesc: "Regenerator + changes type for super effectiveness against opponent.",
+		name: "Guardian Angel",
+		onSwitchOut(pokemon) {
+			pokemon.heal(pokemon.baseMaxhp / 3);
+		},
+		onStart(pokemon) {
+			let resistances = [];
+			Object.keys(Dex.data.TypeChart).forEach(function(type) {
+				if (Dex.getEffectiveness(type, pokemon.side.foe) < 1) return resistances.push(type);
+				return null;
+			}
```

Comment:
```suggestion
			const resistances: string[] = [];
			for (const type of Object.keys(this.dex.data.TypeChart)) {
				if (this.dex.getEffectiveness(type, pokemon.side.foe) >= 1) continue;
				resistances.push(type);
			}
```

this code is still wrong to what the ability is supposed to do, but here's a more optimal way to do what you were doing